risc0/circuit/rv32im/src/lib.rs

```diff
@@ -54,3 +57,35 @@ impl TapsProvider for CircuitImpl {
 }
 
 impl CircuitCoreDef<BabyBear> for CircuitImpl {}
+
+/// Fetch a control ID with the given hash, by name, and cycle limit as a power of two (po2) from
+/// the precomputed table. If the hash function is not precomputed, or the po2 is out of range,
+/// this function will return `None`.
+///
+/// Supported values for hash_name are "sha-256", "poseidon2", and "blake2b".
+#[inline]
+pub fn control_id(hash_name: impl AsRef<str>, po2: usize) -> Option<Digest> {
+    if !(risc0_zkp::MIN_CYCLES_PO2..=risc0_zkp::MAX_CYCLES_PO2).contains(&po2) {
+        return None;
+    }
+    let idx = po2 - risc0_zkp::MIN_CYCLES_PO2;
+    match hash_name.as_ref() {
+        "sha-256" => Some(SHA256_CONTROL_IDS[idx]),
+        "poseidon2" => Some(POSEIDON2_CONTROL_IDS[idx]),
+        "blake2b" => Some(BLAKE2B_CONTROL_IDS[idx]),
+        _ => None,
+    }
+}
+
+/// Fetch all precomputed control IDs using the given hash, by name, and up to the cycle limit as a
+/// power of two (po2). If po2 is larger than the max supported, only supported po2s will be
+/// returned.
+pub fn control_ids(
```

Comment:
I prefer `control_id` and `control_ids` personally. Distinction should be clear enough from the context it's used in, and I try to avoid naming methods with `get`.